src/store/participantsStore.js

```diff
@@ -1036,8 +1093,19 @@ const actions = {
 		}
 	},
 
-	setSpeaking(context, { token, attendeeId, speaking }) {
-		context.commit('setSpeaking', { token, attendeeId, speaking })
+	setSpeakingTime(context, { token, attendeeId, speaking }) {
+		let interval
+
+		if (speaking) {
+			interval = setInterval(() => {
+				context.commit('updateTimeSpeaking', { token, attendeeId })
+			}, 1000)
+		} else {
+			interval = null
+		}
+
+		context.commit('setSpeakingTime', { token, attendeeId, speaking })
+		context.commit('setSpeakingInterval', { token, attendeeId, interval })
```

Comment:
```suggestion
		const interval= speaking ? setInterval(() => {
			context.commit('updateTimeSpeaking', { token, attendeeId })
		})
		: null
		context.commit('setSpeakingTime', { token, attendeeId, speaking })
		context.commit('setSpeakingInterval', { token, attendeeId, interval })
```

Reply:
So, the changes are meant to reduce the redundancy of clearing interval and make each mutation specified to a purpose/object.